app/bundles/LeadBundle/Model/DoNotContact.php

```diff
@@ -74,25 +74,20 @@ public function removeDncForContact($contactId, $channel, $persist = true, $reas
     /**
      * Create a DNC entry for a lead.
      *
-     * @param int          $contactId
-     * @param string|array $channel                  If an array with an ID, use the structure ['email' => 123]
-     * @param string       $comments
-     * @param int          $reason                   Must be a class constant from the DoNotContact class
-     * @param bool         $persist
-     * @param bool         $checkCurrentStatus
-     * @param bool         $allowUnsubscribeOverride
+     * @param string|array $channel If an array with an ID, use the structure ['email' => 123]
+     * @param int          $reason  Must be a class constant from the DoNotContact class
      *
      * @return bool|DNC If a DNC entry is added or updated, returns the DoNotContact object. If a DNC is already present
      *                  and has the specified reason, nothing is done and this returns false
      */
     public function addDncForContact(
-        $contactId,
+        int $contactId,
```

Comment:
This might be a breaking change for anyone who's using this function e.g. in plugins and is not already providing the right data types.

I'm happy to move this snippet to a separate PR; it could've prevented this bug from occurring in the first place, but it's rather a code optimization than part of this bug fix.

Reply:
If it's a BC break and the rest of the bug can be fixed without it, then we should move this part to a separate PR I believe.